lib/active_model/serializer.rb

```diff
@@ -177,6 +177,10 @@ def json_key
       end
     end
 
+    def as_json
+      self.class.adapter.new(self).as_json
```

Comment:
Interesting. Well that certainly makes sense.  I think associations use an adapter, though... Is that a problem for serializing an object snd posting in  a request?

B mobile phone

> On Jun 12, 2015, at 4:23 PM, Ryan Schlesinger <notifications@github.com> wrote:
> 
> In lib/active_model/serializer.rb:
> 
> > @@ -177,6 +177,10 @@ def json_key
> >        end
> >      end
> >  
> > +    def as_json
> > +      self.class.adapter.new(self).as_json
> Agreed. We also use serializers to talk to Elasticsearch outside of a controller.
> 
> —
> Reply to this email directly or view it on GitHub.
>